spec/lib/services/pollers-api-service-spec.js

```
@@ -0,0 +1,255 @@
+                id: "4532",
+                name: "Pollers.IPMI",
+                config: {}
+            }];
+
+            waterline.workitems.find.resolves(mockPoller);
+            return pollerService.getPollers().then(function (pollers) {
+                expect(pollers).to.deep.equal(mockPoller);
+            });
+        });
+        it('should return error if poller informations is not found', function () {
+            var mockPollerError = new Errors.NotFoundError("Could not find workitem with identifier");
+            waterline.workitems.find.rejects(mockPollerError);
+            return pollerService.getPollers().should.eventually.be.rejectedWith(mockPollerError);
+        });
+
+    });
+
+    describe("getPollersById", function() {
+        it("should expose the appropriate methods", function() {
+            pollerService.should.have.property("getPollersById")
+            .that.is.a("function").with.length(1);
+        });
+
+        it("Run getPollersById", function() {
+            var mockPoller = [{
+                id: "1234",
+                name: "Pollers.IPMI",
+                config: {}
+            },
+            {
+                id: "4532",
+                name: "Pollers.TEST",
+                config: {}
+            }];
+            waterline.workitems.needByIdentifier.withArgs("4532").resolves(mockPoller[0]);
+            waterline.workitems.needByIdentifier.withArgs("1234").resolves(mockPoller[1]);
+            Promise.map(["4532", "1234"], function(id) {
+                return pollerService.getPollersById(id);
+            }).then(function (pollers) {
+                _.forEach(pollers, function(poller, index) {
+                    expect(poller).to.deep.equal(mockPoller[index]);
+                });
+            });
+         });
+
+        it("should return error if specific poller info is not found", function () {
+            var mockPollerError = new Errors.NotFoundError("Could not find workitem with identifier");
```

Ignoring this error.